### PR TITLE
[SofaNG] Dispatch SofaHaptics and SofaValidation

### PR DESCRIPTION
**Pull request by fredroy**
https://github.com/sofa-framework/sofa/issues/1527

Deprecates **SofaHaptics** and **SofaValidation** (and a bit of **SofaGeneralObjectInteraction**)

**SofaHaptics** was more or less renamed to Sofa.Component.Haptic
**SofaValidation** only has 2 useful components (CompareState and CompareTopology) which have been moved to S.C.Playback (along the Read and Write ones)
**SofaGeneralObjectInteraction** had an interactiveEllipsoidFF which has been moved to S.C.MechanicalLoad.

The remaining components in the modules/ are now either deprecated (gravity), bogus (monitor&co, collisiongroupmanager) or could be replaced with two components (boxstiffFF)

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/Playback/CMakeLists.txt [hunk 1]
```diff
@@ -6,31 +6,35 @@ set(SOFACOMPONENTPLAYBACK_SOURCE_DIR "src/sofa/component/playback")
...
-    ${SOFACOMPONENTPLAYBACK_SOURCE_DIR}/WriteState.cpp
     ${SOFACOMPONENTPLAYBACK_SOURCE_DIR}/ReadTopology.cpp
-    ${SOFACOMPONENTPLAYBACK_SOURCE_DIR}/InputEventReader.cpp
+    ${SOFACOMPONENTPLAYBACK_SOURCE_DIR}/WriteState.cpp
     ${SOFACOMPONENTPLAYBACK_SOURCE_DIR}/WriteTopology.cpp
 )
 
 sofa_find_package(ZLIB REQUIRED)
-sofa_find_package(Sofa.Simulation.Core REQUIRED)
+sofa_find_package(Sofa.Simulation.Common REQUIRED)
```
Comment (guparan) [inline Sofa/Component/Playback/CMakeLists.txt:34]: PlaybackConfig.cmake.in must be changed accordingly.

File: modules/SofaGeneralObjectInteraction/CMakeLists.txt [hunk 2]
```diff
@@ -15,26 +15,24 @@ set(SOURCE_FILES
...
 
-message(WARNING "${PROJECT_NAME} module is deprecated. It will be removed at v23.06. Use Sofa.Component.SolidMechanics.Spring and Sofa.Component.Constraint.Projective instead.")
+message(WARNING "${PROJECT_NAME} module is deprecated. It will be removed at v23.06. Use Sofa.Component.SolidMechanics.Spring, Sofa.Component.Constraint.Projective and Sofa.Component.MechanicalLoad instead.")
 
 # forward to the new sofang module
 sofa_find_package(Sofa.Component.SolidMechanics.Spring REQUIRED)
 sofa_find_package(Sofa.Component.Constraint.Projective REQUIRED)
-sofa_find_package( Sofa.Component.Engine.Select REQUIRED)
+sofa_find_package(Sofa.Component.MechanicalLoad REQUIRED)
+sofa_find_package(Sofa.Component.Engine.Select REQUIRED)
```
Comment (guparan) [inline modules/SofaGeneralObjectInteraction/CMakeLists.txt]:
```suggestion
sofa_find_package(Sofa.Component.Engine.Select REQUIRED)
sofa_find_package(Sofa.Component.MechanicalLoad REQUIRED)
```

File: modules/SofaGeneralObjectInteraction/SofaGeneralObjectInteractionConfig.cmake.in [hunk 3]
```diff
@@ -5,12 +5,13 @@
...
 if(NOT @PROJECT_NAME@_SENT_DEPRECATION_MESSAGE)
-    message(WARNING "@PROJECT_NAME@ module is deprecated. It will be removed at v23.06. Use Sofa.Component.SolidMechanics.Spring and Sofa.Component.Constraint.Projective instead.")
+    message(WARNING "@PROJECT_NAME@ module is deprecated. It will be removed at v23.06. Use Sofa.Component.SolidMechanics.Spring, Sofa.Component.Constraint.Projective and Sofa.Component.MechanicalLoad instead.")
 endif()
 set_property(GLOBAL PROPERTY PROPERTY_@PROJECT_NAME@_SENT_DEPRECATION_MESSAGE TRUE)
 
 find_package(Sofa.Component.SolidMechanics.Spring QUIET REQUIRED)
 find_package(Sofa.Component.Constraint.Projective QUIET REQUIRED)
+find_package(Sofa.Component.Constraint.Projective QUIET REQUIRED)
 find_package(Sofa.Component.Engine.Select QUIET REQUIRED)
```
Comment (guparan) [inline modules/SofaGeneralObjectInteraction/SofaGeneralObjectInteractionConfig.cmake.in]:
```suggestion
find_package(Sofa.Component.Engine.Select QUIET REQUIRED)
find_package(Sofa.Component.MechanicalLoad QUIET REQUIRED)
```

File: modules/SofaHaptics/CMakeLists.txt [hunk 4]
```diff
@@ -4,37 +4,23 @@ project(SofaHaptics LANGUAGES CXX)
...
 
-if (NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
-    # Silence attribute warnings (for example, ignored already defined external template)
-    target_compile_options(${PROJECT_NAME} PRIVATE -Wno-attributes)
-endif()
+message(WARNING "${PROJECT_NAME} module is deprecated. It will be removed at v23.06. Use Sofa.Component.Haptic instead.")
+
+# forward to the new sofang module
+sofa_find_package(Sofa.Component.Haptic REQUIRED)
+target_link_libraries(${PROJECT_NAME} PUBLIC Sofa.Component.Haptic)
```
Comment (guparan) [inline modules/SofaHaptics/CMakeLists.txt]: SofaHapticsConfig.cmake.in to be changed accordingly.

File: modules/SofaValidation/CMakeLists.txt [hunk 5]
```diff
@@ -48,6 +44,12 @@ sofa_find_package(Sofa.Component.Playback REQUIRED)
 add_library(${PROJECT_NAME} SHARED ${HEADER_FILES} ${SOURCE_FILES})
 target_link_libraries(${PROJECT_NAME} PUBLIC Sofa.Simulation.Common Sofa.Component.Collision Sofa.Component.Playback)
 
+message(WARNING "${PROJECT_NAME} module is deprecated. It will be removed at v23.06. For CompareState and CompareTopology, use Sofa.Component.Playback instead.")
+
+# forward to the new sofang module
+sofa_find_package(Sofa.Component.Playback REQUIRED)
+target_link_libraries(${PROJECT_NAME} PUBLIC Sofa.Component.Playback)
```
Comment (guparan) [inline modules/SofaValidation/CMakeLists.txt:50]: SofaValidationConfig.cmake.in to be changed accordingly.

**Review by guparan — CHANGES_REQUESTED** [review]

**Comment by guparan** [comment]
[ci-build][with-all-tests][force-full-build]

**Comment by guparan** [comment]
[ci-build][with-all-tests][force-full-build]

**Pull request merged**